### PR TITLE
Added document creation

### PR DESCRIPTION
**Pull request by Bretis2019**
Added dashboard page where users can see their rooms, create new rooms and join other user's rooms.

image

### CODE REVIEW

**Comment by strlrd-29** [comment]
Super nice! can you add steps for running it locally, and all env vars required to run it. I got multiple errors when trying to run it.

**Comment by Bretis2019** [comment]
to run locally:
1- clone https://github.com/Bretis2019/Hiring-software-engineer-frontend.git
2- switch  Added document creation #9  branch
3- cd Bigmama
4- create .env.local file with the following variables
`LIVEBLOCKS_SECRET_KEY`
`NEXT_PUBLIC_SUPABASE_URL`
`NEXT_PUBLIC_SUPABASE_ANON_KEY`
5- npm install
6- npm run dev

**Comment by strlrd-29** [comment]
Great I will check it out.

**Comment by Bretis2019** [comment]
### Updated dashboard cards
image
### Added shadcn shelf for creating rooms
Screen Shot 2023-12-31 at 00 22 00
### Added share button to room toolbar
image
### Replaced magnet link with google sign in
image

**Comment by Bretis2019** [comment]
Any feedback would be appreciated !

**Comment by strlrd-29** [comment]
Thank you for your contributions @Bretis2019, I will be checking it out.

**Comment by strlrd-29** [comment]
@Bretis2019 I wasn't able to sign up, can you provide steps on how to do it. Plus how to enable the google and discord auth from supabase.

File: Bigmama/src/components/auth-form.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+'use client'
+import { Auth } from '@supabase/auth-ui-react'
+import { ThemeSupa } from '@supabase/auth-ui-shared'
+import { createClientComponentClient } from '@supabase/auth-helpers-nextjs'
+import { useEffect } from "react";
+import { useRouter } from "next/navigation";
+import { useLottie } from "Lottie-react";
```
Comment (strlrd-29) [inline Bigmama/src/components/auth-form.tsx:7]: There is a typo in here, it is supposed to be `lottie-react`.

**Comment by Bretis2019** [comment]
## Database Setup

Create a Supabase project at Supabase Dashboard.

Go to the SQL Editor in your Supabase project and execute the following SQL queries:

```sql
  -- Create a table for public profiles
CREATE TABLE profiles (
  id UUID REFERENCES auth.users ON DELETE CASCADE NOT NULL PRIMARY KEY,
  updated_at TIMESTAMP WITH TIME ZONE,
  username TEXT UNIQUE,
  full_name TEXT,
  avatar_url TEXT,
  website TEXT,
  profile_color VARCHAR(7),
  rooms TEXT[],  -- Adding an array of strings for rooms
  CONSTRAINT username_length CHECK (CHAR_LENGTH(username) >= 3)
);

-- Set up Row Level Security (RLS)
ALTER TABLE profiles
  ENABLE ROW LEVEL SECURITY;

CREATE POLICY "Public profiles are viewable by everyone." ON profiles
  FOR SELECT USING (TRUE);

CREATE POLICY "Users can insert their own profile." ON profiles
  FOR INSERT WITH CHECK (auth.uid() = id);

CREATE POLICY "Users can update own profile." ON profiles
  FOR UPDATE USING (auth.uid() = id);

-- This trigger automatically creates a profile entry when a new user signs up via Supabase Auth.
CREATE FUNCTION public.handle_new_user()
RETURNS TRIGGER AS $$
BEGIN
  INSERT INTO public.profiles (id, full_name, avatar_url, rooms)
  VALUES (NEW.id, NEW.raw_user_meta_data->>'full_name', NEW.raw_user_meta_data->>'avatar_url', ARRAY[]::TEXT[]);
  RETURN NEW;
END;
$$ LANGUAGE plpgsql SECURITY DEFINER;

CREATE TRIGGER on_auth_user_created
  AFTER INSERT ON auth.users
  FOR EACH ROW EXECUTE PROCEDURE public.handle_new_user();

-- Set up Storage!
INSERT INTO storage.buckets (id, name)
  VALUES ('avatars', 'avatars');

-- Set up access controls for storage.
CREATE POLICY "Avatar images are publicly accessible." ON storage.objects
  FOR SELECT USING (bucket_id = 'avatars');

CREATE POLICY "Anyone can upload an avatar." ON storage.objects
  FOR INSERT WITH CHECK (bucket_id = 'avatars');

```




## Set up Authentication Providers



**Discord Authentication**

1. In the "Authentication" tab, find the "Discord" provider.

2. Enable the Discord provider.

3. Head over to the [Discord Developer Portal](https://discord.com/developers/applications) and create a new application.

4. In the Oauth2 tab, copy the client ID and secret back to Supabase.

5. From Supabse copy Callback URL and paste in the discord Oauth2 tab.

**Google Authentication**

Mainly the same steps as Discord but feel free to check out [this](https://www.youtube.com/watch?v=_XM9ziOzWk4) tutorial from Supabase.

**Comment by Bretis2019** [comment]
You can try it [here](https://bigmama.vercel.app)

NOTE: I'm still fixing some bugs with login with email and password but social logins should work just fine.
If there's something you'd like me to fix or change leave a comment and I'll get right on it.

**Comment by strlrd-29** [comment]
Good job @Bretis2019, I will be closing this pull request and we will be contacting you for an in person interview next week Inchallah.

**Pull request closed without merging**